src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -106,7 +99,10 @@ class MSIAuthenticationWrapper(MSIAuthentication):
     # This method is exposed for Azure Core. Add *scopes, **kwargs to fit azure.core requirement
     def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         logger.debug("MSIAuthenticationWrapper.get_token invoked by Track 2 SDK with scopes=%s", scopes)
-        self.resource = scopes_to_resource(scopes)
+        resource = _try_scopes_to_resource(scopes)
```

Comment:
If the resource is None, which resource is used to get token then?

Reply:
The `resource` used to create `MSIAuthenticationWrapper` will be used.

https://github.com/Azure/azure-cli/blob/1b2786784b57a9f3de12a15de92804f47f11ad0e/src/azure-cli-core/azure/cli/core/_profile.py#L802